File: Frontend/AppFavoresEscritorio/AppFavoresEscritorio/Modelos/Autenticacion/RespuestaLogin.cs
using System.Text.Json.Serialization;

namespace AppFavoresEscritorio.Modelos.Autenticacion;

public class RespuestaLogin
{
    [JsonPropertyName("access_token")]
    public string AccessToken { get; set; } = string.Empty;

    [JsonPropertyName("token_type")]
    public string TokenType { get; set; } = string.Empty;

    [JsonPropertyName("usuario")]
    public UsuarioResumen? Usuario { get; set; }
}

public class UsuarioResumen
{
    [JsonPropertyName("nombre")]
    public string Nombre { get; set; } = string.Empty;

    [JsonPropertyName("correo")]
    public string Correo { get; set; } = string.Empty;

    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;
    
    [JsonPropertyName("rol")]
    public string Rol { get; set; } = string.Empty;
}

File: Frontend/AppFavoresMovil/app/src/main/java/com/example/appfavoresmovil/data/modelos/autenticacion/RespuestaLogin.kt
package com.example.appfavoresmovil.data.modelos.autenticacion

import com.google.gson.annotations.SerializedName

data class RespuestaLogin(
    @SerializedName("access_token") val accessToken: String,
    @SerializedName("token_type") val tokenType: String,
    @SerializedName("usuario") val usuario: UsuarioResumen?
)

data class UsuarioResumen(
    @SerializedName("nombre") val nombre: String,
    @SerializedName("correo") val correo: String,
    @SerializedName("id") val id: String,
    @SerializedName("rol") val rol: String
)
